libs/s25main/factories/BuildingFactory.cpp

```diff
@@ -42,6 +43,17 @@ noBuilding* BuildingFactory::CreateBuilding(GameWorldBase& gwg, const BuildingTy
         case BLD_SHIPYARD: bld = new nobShipYard(pt, player, nation); break;
         default: bld = new nobUsual(type, pt, player, nation); break;
     }
+    GamePlayer &player_obj = gwg.GetPlayer(player);
```

Comment:
Could you move this either below the `AddBuilding` (and reuse player_obj, btw: `lowerCamelCase` please) or into `GamePlayer::AddBuilding`? I think the latter is the better place ( I don't like doing to much in the factory, actually it should just create the building and nothing else)